### DISCUSSION TITLE
how to hide banner

### DISCUSSION THREAD

**Discussion by Nuesret** [admob-plus/admob-plus#692] [Q&A]
I'm using the admob-plus plugin in a cordova project in Android Studio. It works fine (I have almost over 10 unitID for over 80 html-pages. I am using different unitIDs for better analyzing purposes in Admob statistics. Everything is working fine. But there are seperate html-pages with embeded Youtube-Content which are now a problem: Google Play rejected the app because the banners are violating their policy. I understand, that if I remove the banner ads from that specific Youtube pages (make them hide/invisible)- everything will be fine. So I tried to hide the banner with this following javascript code in the header of the specific html pages with this admob-plus banner code...

let banner

document.addEventListener('deviceready', async () => {
  banner = new admob.BannerAd({
    adUnitId: 'ca-app-pub-xxx/yyy',
  })

  banner.on('impression', async (evt) => {
    await banner.hide()
  }, true)

  await banner.show()
}, false)

But the banner ist still visible at the bottom - floating in superview over the content. As you can see, I added after await banner.hide() a true... but this isn't the solution. I also removed the last part with await.banner.show() but the banner ist still visible. 

I understand, that I should use the same unitID of the banner, which I want to hide. So maybe I should use only one unitID for all pages to make it easier to hide the banners, right? But what exactly am I doing wrong in the hide code. Can you help?

**Comment by EMI-INDO** [comment]
try

```
if (banner){

await banner.hide()

}
```

**Comment by Nuesret** [comment]
Thank you for your feedback. No, it didn't work. I'm doing it wrong. 

```
<script>
if (banner){
await banner.hide()
}
</script>
```

...I guess, I'm doing it wrong. 

Here is my situation:

I call the banner for the first time on index.html

```
<script>
let banner
document.addEventListener('deviceready', async () => {
banner = new admob.BannerAd({
adUnitId: 'ca-app-pub-3940256099942544/6300978111',
})
await banner.show()
}, true)
</script>
```

Ok. The banner is displayed immediately and without delay. The aim is that the banner remains visible on the following pages and is hidden on individual pages (with YouTube content).

If I place on each page in the header

```
<script>
banner.on('impression', async (evt) => {
    await banner.show()
  })
</script>
```

...then the banner is showing up. But each time the index page is called up again, the same banner is built up and is placed over the old one - until the page/content is covered with the banner duplicates at some point.

What am I doing wrong?
What should the script look like so that the banner is not duplicated when the page is visited again? 
And what does the script have to look like in order to hide the banner? 

Is it even possible to do this directly with javascript in the header?

**Comment by Nuesret** [comment]
Found a possible solution made in 2022 by alejandroicod  

> Basically I create a new banner using the last Id that was used. In case there is already a banner with that Id it gets destroyed, catch the rejection of the promise and create a new one with Id null (which will force Admob to generate the next available Id). This way I always make sure there is only one banner visible at a time.

I'm testing his work around 
```
function initBanner(id) {
  let banner = new admob.BannerAd({
    adUnitId: 'ca-app-pub-xxxxxxxx' ,
    position: 'bottom',
    npa: status,
    id: id ? Number.parseInt(id) : null
  });

  banner.on('load', (ex) => {
    localStorage.setItem('lastBannerId', ex.adId);
  })

  return banner.show();
}

const lastBannerId = localStorage.getItem('lastBannerId');

await initBanner(lastBannerId)
  .catch(ex => {
    initBanner(null)
  });
```

**Comment by Nuesret** [comment]
Argh! It's not working... what am I doing wrong?

This is the situation:

1. made my webproject in html-css-javascript
2. build the corona project and added all components: the admob-plus plugin and androidx adapters
3. imported the android folder in Android Studio, synced and added the necessary meta.data in manifest.xml
4. implemented the abdmob code in MainActivity.java and checked gradle.build for actual google ads lib
5. added the let banner snippets to the html headers in assets
```
let banner

document.addEventListener('deviceready', async () => {
banner = new admob.BannerAd({
adUnitId: 'ca-app-pub-xxx/yyy',
})

banner.on('impression', async (evt) => {
await banner.hide()
})

await banner.show()
}, false)
```
Result: the banner is jumping in and out and stays after an amount of time (at least, if you revisit the page again). 
If I change the code in
```
let banner

document.addEventListener('deviceready', async () => {
banner = new admob.BannerAd({
adUnitId: 'ca-app-pub-xxx/yyy',
})

banner.on('impression', async (evt) => {
await banner.hide()
})

await banner.show()
}, true)
```
...the banner is showing up and stays. BUT... this works fine as long as the user opens the pages in a linear way. If he revisits the page - the same banner duplicates itself and jump over the old one: now we see two identical banners at the bottom. This goes on and on... till the whole page is covered with the same banner.

Hey guys, how do you manage it to work? Did I forget to add code somewhere? Please help. I need really help.

Found this but it isn't working
[https://github.com/admob-plus/admob-plus/issues/450](url)
